api/search.rb

```diff
@@ -125,14 +125,21 @@ def get_threads(context, group_ids, local_params, search_text)
   result_obj.to_json
 end
 
+error Sinatra::Param::InvalidParameterError do
+  # NOTE (CCB): The current behavior of the service is to return a seemingly positive response
+  # for an invalid request. In the future the API's contract should be modified so that HTTP 400
+  # is returned. This informs the client that the request was invalid, rather than having to guess
+  # about an empty response body.
+  [200, '{}']
+end
+
 get "#{APIPREFIX}/search/threads" do
+  param :text, String, required: true
+  param :context, String, default: 'course'
+  param :sort_order, String, in: %w(asc desc), transform: :downcase
```

Comment:
Also, in the past what did it do if you provided an illegal sort_key?  Did it give an error?  Looks like the new code will give an empty result for all errors, which may not have been true in the past.  Since this is something we don't really want, we shouldn't make the empty result apply to more situations than it needs to.